src/main/jbake/content/blog/2017/01/16/hawkular-metrics-with-dropwizard.adoc

```diff
@@ -0,0 +1,339 @@
+
+If, for instance, we want to connect to an Hawkular pod running in OpenShift, we would configure the reporter similar to that one:
+``` java
+    HawkularReporter hawkularReporter = HawkularReporter.builder(registry, "com.hawkular.sample")
+            .uri("https://metrics.192.168.42.63.xip.io/hawkular/metrics")
+            .bearerToken("ABCDEFGHIJ1234")
+            .build();
+```
+
+To conclude with the initial setup, note that it is often a good practice to include some host-based disambiguation string in the metric names,
+so that if your application is deployed on several hosts and they all communicate to the same Hawkular instance using the same tenant,
+they will not conflict and feed the same metrics.
+A convenient way to do that is to set a prefix in the reporter configuration, for instance using the host name:
+
+``` java
+    String hostname;
+    try {
+        hostname = InetAddress.getLocalHost().getCanonicalHostName();
+    } catch (UnknownHostException e) {
+        hostname = "?";
+    }
+    HawkularReporter hawkularReporter = HawkularReporter.builder(registry, "com.hawkular.sample")
+            .prefixedWith(hostname + ".")
+            .build();
+```
+
+=== Dropwizard -> Hawkular
+
+As stated before, our sample app will do some (very) basic benchmarking on several caches implementations: we'll have a _Guava_ cache and a local _EhCache_.
+
+NOTE: This sample app is on my GitHub: https://github.com/jotak/hawkular-dropwizard-sample
+
+We create a very simple interface that allows us to get data from cache, count number of elements and know if the latest read was from cache or not:
+
+``` java
+interface Backend {
+    Object get(String key);
+    long count();
+    void init(Map<String, Object> presetElements);
+    boolean isLastReadFromCache();
+}
+```
+
+You can link:https://github.com/jotak/hawkular-dropwizard-sample/tree/master/src/main/java/com/hawkular/sample[see on GitHub] the `GuavaBackend` and `EhcacheBackend` implementations, there's nothing fancy here.
+
+EhCache is initialized programmatically:
+
+``` java
+    CacheManager cacheManager = CacheManager.newInstance();
+    Ehcache cache = cacheManager.addCacheIfAbsent("testCache");
+```
+
+Now, the `Benchmark` class. This is where we create metrics and feed them, with the pretext of a small scenario. Here's the `run` method:
+
+``` java
+    private void run() {
+        final DatabaseStub fakeDb = new DatabaseStub(); // <1>
+        final BackendMonitoring monitoring = new BackendMonitoring(registry); // <2>
+        Map<String, Object> presetElements = IntStream.range(0, 100000)
+                .mapToObj(Integer::new)
+                .collect(Collectors.toMap(i -> UUID.randomUUID().toString(), i -> i)); // <3>
```

Comment:
(minor thing)
instead of
```java
                 .mapToObj(Integer::new)
                 .collect(Collectors.toMap(i -> UUID.randomUUID().toString(), i -> i));
```

isn't it better to do just
```java
                  .collect(Collectors.toMap(i -> UUID.randomUUID().toString(), i -> new Integer(i)));
```